pkg/otlp/metrics/metrics_remapping.go

```diff
@@ -0,0 +1,194 @@
+	m.SetName("otel." + m.Name())
+}
+
+// remapContainerMetrics extracts system metrics from m and appends them to all.
+func remapContainerMetrics(all pmetric.MetricSlice, m pmetric.Metric) {
+	name := m.Name()
+	if !strings.HasPrefix(name, "container.") {
+		// not a container metric
+		return
+	}
+	switch name {
+	case "container.cpu.usage.total":
+		if addm, ok := copyMetric(all, m, "container.cpu.usage", 1); ok {
+			addm.SetUnit("nanocore")
+		}
+	case "container.cpu.usage.usermode":
+		if addm, ok := copyMetric(all, m, "container.cpu.user", 1); ok {
+			addm.SetUnit("nanocore")
+		}
+	case "container.cpu.usage.system":
+		if addm, ok := copyMetric(all, m, "container.cpu.system", 1); ok {
+			addm.SetUnit("nanocore")
+		}
+	case "container.cpu.throttling_data.throttled_time":
+		copyMetric(all, m, "container.cpu.throttled", 1)
+	case "container.cpu.throttling_data.throttled_periods":
+		copyMetric(all, m, "container.cpu.throttled.periods", 1)
+	case "container.memory.usage.total":
+		copyMetric(all, m, "container.memory.usage", 1)
+	case "container.memory.active_anon":
+		copyMetric(all, m, "container.memory.kernel", 1)
+	case "container.memory.hierarchical_memory_limit":
+		copyMetric(all, m, "container.memory.limit", 1)
+	case "container.memory.usage.limit":
+		copyMetric(all, m, "container.memory.soft_limit", 1)
+	case "container.memory.total_cache":
+		copyMetric(all, m, "container.memory.cache", 1)
+	case "container.memory.total_swap":
+		copyMetric(all, m, "container.memory.swap", 1)
+	case "container.blockio.io_service_bytes_recursive":
+		copyMetric(all, m, "container.io.write", 1, kv{"operation", "write"})
+		copyMetric(all, m, "container.io.read", 1, kv{"operation", "read"})
+	case "container.blockio.io_serviced_recursive":
+		copyMetric(all, m, "container.io.write.operations", 1, kv{"operation", "write"})
+		copyMetric(all, m, "container.io.read.operations", 1, kv{"operation", "read"})
+	case "container.network.io.usage.tx_bytes":
+		copyMetric(all, m, "container.net.sent", 1)
+	case "container.network.io.usage.tx_packets":
+		copyMetric(all, m, "container.net.sent.packets", 1)
+	case "container.network.io.usage.rx_bytes":
+		copyMetric(all, m, "container.net.rcvd", 1)
+	case "container.network.io.usage.rx_packets":
+		copyMetric(all, m, "container.net.rcvd.packets", 1)
+	}
+}
+
+// kv represents a key/value pair.
+type kv struct{ K, V string }
+
+// copyMetric copies metric m to dest. The new metric's name will be newname, and all of its datapoints will
+// be divided by div. If filter is provided, only the data points that have *either* of the specified string
+// attributes will be copied over. If the filtering results in no datapoints, no new metric is added to dest.
+//
+// Please note that copyMetric is restricted to the metric types Sum and Gauge.
+func copyMetric(dest pmetric.MetricSlice, m pmetric.Metric, newname string, div float64, filter ...kv) (pmetric.Metric, bool) {
```

Comment:
May also want to document the returned `bool`

Reply:
Indeed. I added a sentence about the return values.